File: test/whileloop.pas
program whileloop;
　　var m,n,a,b,r:integer;
　　begin
　　　　write("Input m,n:");
　　　　read(m,n);
　　　　a:=m;b:=n;r:=a mod b;
　　　　while r<>0 do
　　　　begin
　　　　　a:=b;b:=r;
　　　　　r:=a mod b;
　　　　end;
　　　　write("The greatest common divide is : ");
       write(b);
　　end.

File: test/whileloop.c
#include <stdio.h>

int m, n, a, b, r;

int main() {
    printf("Input m,n:");
    scanf("%d %d", &m, &n);
    a = m;
    b = n;
    r = a % b;
    while (r != 0) 
    {
        a = b;
        b = r;
        r = a % b;
    }
    printf("The greatest common divide is : ");
    printf("%d", b);
    return 0;
}